### ISSUE TITLE
Visualization of Propagation Matrix for AGNN

### ISSUE THREAD

**Issue by rayush7** [rusty1s/pytorch_geometric#71]
Hi @rusty1s 

I have a question regarding Attention Based Graph Neural Networks for Semi Supervised Learning i.e AGNN layer. How can I visualize the propagation matrix of various AGNN layers once the training has been completed? Could you please give an example for the CORA dataset?

Thanks

**Comment by rusty1s** [comment]
Hi,
if I understand you correctly, you want to visualize `P_ij` (according to the [paper](https://arxiv.org/pdf/1803.03735.pdf), so you need a way to obtain this sparse matrix before multiplication with node features.

Currently, this is not possible, but I can outsource the building of the propagation matrix to its own method. WAYT?

**Comment by rusty1s** [comment]
The `AGNNConv` now implements the method `propagation_matrix(x, edge_index)`, which returns the propagation matrix `(edge_index, edge_weight)` with inserted self-loops and attention-based edge weights.

**Comment by rayush7** [comment]
Thank you so much @rusty1s. This is exactly what I wanted. The propagation matrix helps to better interpret the results than just the accuracy achieved by the model.
